### DISCUSSION TITLE
What is the difference between using constraint_func and manually penalized objective function

### DISCUSSION THREAD

**Discussion by lucidfrontier45** [optuna/optuna#5757] [Q&A]
I want to know how those approaches work differently in Optuna and which is the recommended one.

1. Using Optuna Sampler's constraint_func feature as in document
2. Let objective function return `inf` if constraints are violated
3. Let trial fail by setting TrialState.FAIL

Thank you

**Comment by rodrigoscoelho** [comment]
I’m also eager to find out the same information. If anyone could clarify this for us, it would be greatly appreciated. Thank you!
